### PR TITLE
Annotate collections.specialized to be native aot compatible

### PR DESCRIPTION
**Pull request by LakshanF**
Also trying out native aot tests in runtime-extra pipeline with linux

### CODE REVIEW

File: src/libraries/System.Collections.Specialized/src/System/Collections/Specialized/NameObjectCollectionBase.cs [hunk 1]
```diff
@@ -431,6 +431,8 @@ bool ICollection.IsSynchronized
         ///    <para>Returns an array of the specified type containing
         ///       all the values in the <see cref='System.Collections.Specialized.NameObjectCollectionBase'/> instance.</para>
         /// </devdoc>
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
+            Justification = "The cast is safe for created array")]
```
Comment (MichalStrehovsky) [inline src/libraries/System.Collections.Specialized/src/System/Collections/Specialized/NameObjectCollectionBase.cs:435]:
```suggestion
            Justification = "The API only works for reference type arguments and code for reference typed arrays is shareable.")]
```

**Comment by LakshanF** [comment]
This PR also enabled linux-arm64 for the RR and it looks like that is [passing](https://dev.azure.com/dnceng/public/_build/results?buildId=1847232&view=logs&j=d6b9692e-ec44-5c72-7bbd-3282d42dbbe2).

**Comment by MichalStrehovsky** [comment]
Cc @dotnet/area-system-collections on the System.Collections.Specialized change. Might want to consider adding an argument check in the future so that we don't go into the InvalidCastException codepath when the [cast to `object[]` fails](https://github.com/dotnet/runtime/pull/71159#discussion_r904404477). But also, I've never in my life heard about this collection type so probably not many people care.

**Review by MichalStrehovsky — APPROVED** [review]

**Pull request merged**